docs/velocity/dev/api/plugin-messaging.md

```diff
@@ -0,0 +1,160 @@
+
+## Case 1: Receiving a Plugin Message from a Player
+
+When you need to handle or inspect the contents of a plugin message sent by a player.
+
+This requires registering with the ChannelRegistrar for the event to be fired.
+
+An example use might be logging messages from a mod that reports what features are enabled.
+
+```java
+public static final MinecraftChannelIdentifier IDENTIFIER = MinecraftChannelIdentifier.from("custom:main");
+
+@Subscribe
+public void onProxyInitialization(ProxyInitializeEvent event) {
+    proxyServer.getChannelRegistrar().register(IDENTIFIER);
+}
+
+@Subscribe
+public void onPluginMessageFromPlayer(PluginMessageEvent event) {
+    if (!(event.getSource() instanceof Player)) {
+        return;
+    }
+    Player player = (Player) event.getSource();
+    // Ensure the identifier is what you expect before trying to handle the data
+    if (event.getIdentifier() != IDENTIFIER) {
+        return;
+    }
+    
+    ByteArrayDataInput in = ByteStreams.newDataInput(event.getData());
+    // handle packet data
+}
+```
+
+
+## Case 2: Sending a Plugin Message to a Backend Server
+
+When you need to send a message to a backend server.
+
+There are two methods to send a plugin message to the backend, depending on what you need to achieve
+
+:::warning
+
+When listening for these messages on a backend server, ensure you are actually talking to an (allowed) instance of your proxy plugin.
+
+If you don't, you run the risk that these can be spoofed by a player with a modded client.
+
+:::
+
+### Using any connected Player
+
+This is useful if you just want to communicate something relevant to the entire server,
+or otherwise can be derived from its content.
+
+
+An example use might be telling the server to shut down.
+
+
+```java
+public boolean sendPluginMessageToBackend(RegisteredServer server, ChannelIdentifier identifier, byte[] data) {
+    boolean success = server.sendPluginMessage(identifier, data);
+    return success;
+}
+```
+
+### Using a specific player's connection
+
+This is useful if you want to communicate something about a specific player to their current backend server.
+You may want additional checks to ensure it will be handled correctly on the backend the player is on.
+
+An example use might be telling the backend server to give the player a specific item.
+
+```java
+public boolean sendPluginMessageToBackendUsingPlayer(Player player, ChannelIdentifier identifier, byte[] data) {
+    Optional<ServerConnection> connection = player.getCurrentServer();
+    boolean success = false;
+    if (connection.isPresent()) {
+        success = connection.get().sendPluginMessage(identifier, data);
+    }
+    return success;
+}
+```
+
+## Case 3: Receiving a Plugin Message from a Backend server
+
+When you need to Receive messages from your backend server.
+
+This requires registering with the ChannelRegistrar for the event to be fired.
+
+An example use might be handing a request to transfer the player to another server.
```

Comment:
```suggestion
This is for when you need to receive messages from your backend server. It will require registering with the ChannelRegistrar for the event to be fired.

An example use might be handing a request to transfer the player to another server.
```